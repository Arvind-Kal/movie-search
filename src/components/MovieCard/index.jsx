import React from 'react';
import styles from './styles/MovieCard.module.scss';
import PropTypes from 'prop-types';

const MovieCard = ({movies}) => {

    const makeMovieCard = (movie) => {
        return (
            <div className={styles.card} key={movie.id}>
                <img 
                className={styles.card__image}
                src={`https://image.tmdb.org/t/p/w500/${movie.poster_path}`}
                alt={movie.title + ' poster'}/>
                <div className={styles.card__content}>
                    <h3 className={styles.card__title}> {movie.title} </h3>
                    <p className={styles.card__info}>Release Date: {movie.release_date} </p>
                    <p className={styles.card__info}>Rating: {movie.vote_average} </p>
                    <p className={styles.card__info}>Synopsis: {movie.overview} </p>
                </div>
            </div>
        );
    };
    return (
        <div className={styles.cardList}>
           {movies.filter(movie => movie.poster_path).map(makeMovieCard)}
        </div>
    );
};

MovieCard.propTypes = {
    /**
     * The list of movies being passed in
     */
    movies: PropTypes.array
};

MovieCard.defaultProps = {
    movies: []
}

export default MovieCard;