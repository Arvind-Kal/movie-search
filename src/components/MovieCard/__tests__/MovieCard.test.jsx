import { render } from '@testing-library/react';
import MovieCard from '../lib';

describe('MovieCard', () => {
    it('should render Movie Card with no movies passed in.', () => {
        const { asFragment } = render(<MovieCard></MovieCard>);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should render Movie Card with a list of movies passed in.', () => {
        const testMovies = [
            {
                title: 'test movie',
                release_date: 'test release date',
                vote_average: 'test review',
                overview: 'test synopsis',
                poster_path: 'test poster path'
            }
        ];
        const { asFragment } = render(<MovieCard movies={testMovies}></MovieCard>);
        expect(asFragment()).toMatchSnapshot();
    });
});