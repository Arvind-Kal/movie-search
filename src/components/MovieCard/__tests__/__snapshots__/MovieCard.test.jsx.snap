// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should render Movie Card with a list of movies passed in. 1`] = `
<DocumentFragment>
  <div
    class="cardList"
  >
    <div
      class="card"
    >
      <img
        alt="test movie poster"
        class="card__image"
        src="https://image.tmdb.org/t/p/w500/test poster path"
      />
      <div
        class="card__content"
      >
        <h3
          class="card__title"
        >
           test movie 
        </h3>
        <p
          class="card__info"
        >
          Release Date: test release date 
        </p>
        <p
          class="card__info"
        >
          Rating: test review 
        </p>
        <p
          class="card__info"
        >
          Synopsis: test synopsis 
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MovieCard should render Movie Card with no movies passed in. 1`] = `
<DocumentFragment>
  <div
    class="cardList"
  />
</DocumentFragment>
`;
